src/Ccap/Codegen/Scala.purs

```diff
@@ -1321,27 +1321,47 @@ encodeTypeParam (Cst.TypeParam t) e =
     <> e
     <> ")"
 
-recordFieldType :: TypeDeclOutputMode -> Ast.RecordProp -> Codegen String
-recordFieldType mode { annots, name, typ } = do
+recordFieldType :: TypeDeclOutputMode -> Boolean -> Ast.RecordProp -> Codegen String
+recordFieldType mode withDefaultArguments { annots, name, typ } = do
   ty <- case typ of
     Ast.TType t -> typeDef mode t
     Ast.TParam (Cst.TypeParam c) -> pure (initialUpper c)
   let
-    -- If a field is marked as DbManaged add a default argument.
     defaultArgument =
-      if Annotations.getIsDbManaged annots then
-        defaultArgumentForType typ
+      if withDefaultArguments && Annotations.getIsDbManaged annots then
+        defaultArgumentForType ty typ
       else
         Nothing
   pure
     $ arg name ty
     <> maybe "" (\dA -> " = " <> dA) defaultArgument
   where
-  defaultArgumentForType = case _ of
+  defaultArgumentForType typeDefinition = case _ of
     Ast.TType tTyp -> case tTyp of
       Ast.Ref tRef ->
-        if tRef.typ == "OccId" then Just "OccSupport.empty" else Nothing
-      _ -> Nothing
+        if tRef.typ == "OccId" then
+          Just "OccSupport.empty"
+        else
+          tRef.decl >>=
+            ( case _ of
+                (Tuple _ (Ast.TypeDecl { topType: Ast.Wrap prim@(Ast.Primitive _) })) ->
+                   map
+                     (\primitiveDefault -> typeDefinition <> "(" <> primitiveDefault <> ")")
+                     (defaultArgumentForType typeDefinition (Ast.TType prim))
```

Comment:
Could this be split out to a separate function (the default value for a primitive type) and then used in both places? typeDefinition effectively isn't used here.

Reply:
Split this out